silq/analysis/analysis.py

```diff
@@ -715,6 +717,38 @@ def analyse_EPR(empty_traces: np.ndarray,
             'mean_high_blip_duration': results_read['mean_high_blip_duration']}
 
 
+def analyse_threshold_up_proportion(up_proportions_arrs: np.ndarray,
+                                    shots_per_frequency: int):
+    proportion_space = np.linspace(0, 1, num=shots_per_frequency + 1,
+                                   endpoint=True)
+    up_proportions_arrs = up_proportions_arrs.reshape(1, -1)
+    kernel = gaussian_kde(up_proportions_arrs)
+    gaussian_up_proportions = kernel(proportion_space)
+    peak_idxs = peakutils.peak.indexes(gaussian_up_proportions,
+                                       thres=0.5/up_proportions_arrs.shape[-1],
+                                       min_dist=shots_per_frequency/5)
+    if len(peak_idxs) == 0:
+        logger.debug(f'Adaptive thresholding routine: 0 peaks were '
+                     f'found, using threshold 0.5')
+        true_idx = round((len(proportion_space) - 1) / 2)
+    elif len(peak_idxs) == 1:
+        if (len(proportion_space) - peak_idxs[0]) / len(proportion_space) > 0.5:
+            true_slice = slice(*[peak_idxs[0], len(proportion_space)])
+            true_idx = np.argmin(np.round(gaussian_up_proportions[true_slice], 4)) + \
+                       peak_idxs[0]
+        else:
+            true_slice = slice(*[0, peak_idxs[0]])
+            true_idx = np.argmin(np.round(gaussian_up_proportions[true_slice], 4))
```

Comment:
I don't think any of this is necessary, if you've found only 1 peak then it doesn't matter where you place the threshold, it could be either at 0 or 1, depending on whether the peak is in the left- or right-half of the proportion space.

Do you have a counter-example?

Reply:
This refers to the same issue above, I want to have threshold oscillating somewhere around 0.5 (in the middle of spin-up proportion space) so that it would have potential application in the future.